File: app/src/main/java/com/icerockdev/babenko/core/dagger/modules/SharedPreferencesModule.java
package com.icerockdev.babenko.core.dagger.modules;

import android.content.Context;

import com.icerockdev.babenko.managers.impl.SharedPreferencesManagerImpl;
import com.icerockdev.babenko.managers.SharedPreferencesManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Roman Babenko on 17/10/17.
 */
@Module
public class SharedPreferencesModule {
    @Singleton
    @Provides
    SharedPreferencesManager provideSharedPreferencesManager(Context context) {
        return new SharedPreferencesManagerImpl(context);
    }
}

File: app/src/main/java/com/icerockdev/babenko/applicaiton/dagger/modules/SharedPreferencesModule.kt
package com.icerockdev.babenko.applicaiton.dagger.modules

import android.content.Context
import com.icerockdev.babenko.managers.SharedPreferencesManager
import com.icerockdev.babenko.managers.impl.SharedPreferencesManagerImpl
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by Roman Babenko on 17/10/17.
 */
@Module
class SharedPreferencesModule {
    @Singleton
    @Provides
    internal fun provideSharedPreferencesManager(context: Context): SharedPreferencesManager {
        return SharedPreferencesManagerImpl(context)
    }
}
